poe-tasks/build-and-publish-java-connectors-with-tag.sh

```diff
@@ -15,55 +29,68 @@
 #
 #   4) Mixed: positional + pre-release
 #   ./build-and-publish-java-connectors-with-tag.sh --pre-release foo-conn
+#
+#   5) Enable actual publishing (default is dry-run mode)
+#   ./build-and-publish-java-connectors-with-tag.sh --publish foo-conn
 set -euo pipefail
 
 CONNECTORS_DIR="airbyte-integrations/connectors"
 
 # ── Rollout whitelist: only connectors listed here will be built/published
-declare -A rollout_map=(
-  [destination-azure-blob-storage]=1
-  [destination-bigquery]=1
-  [destination-clickhouse-strict-encrypt]=1
-  [destination-clickhouse]=1
-  [destination-csv]=1
-  [destination-databricks]=1
-  [destination-dev-null]=1
-  [destination-dynamodb]=1
-  [destination-elasticsearch-strict-encrypt]=1
-  [destination-elasticsearch]=1
-  [destination-gcs]=1
-  [destination-kafka]=1
-  [destination-local-json]=1
-  [destination-mongodb-strict-encrypt]=1
-  [destination-mongodb]=1
-  [destination-mysql-strict-encrypt]=1
-  [destination-mysql]=1
-  [destination-oracle-strict-encrypt]=1
-  [destination-oracle]=1
-  [destination-postgres-strict-encrypt]=1
-  [destination-postgres]=1
-  [destination-redis]=1
-  [destination-redshift]=1
-  [destination-s3-data-lake]=1
-  [destination-s3]=1
-  [destination-singlestore]=1
-  [destination-snowflake]=1
-  [destination-starburst-galaxy]=1
-  [destination-teradata]=1
-  [destination-yellowbrick]=1
-  [source-e2e-test]=1
-  [source-postgres]=1
-  [source-mysql]=1
-)
+# Function to check if a connector is in the whitelist
+is_in_whitelist() {
+  local connector="$1"
+  case "$connector" in
+    destination-azure-blob-storage|\
+    destination-bigquery|\
+    destination-clickhouse-strict-encrypt|\
+    destination-clickhouse|\
```

Comment:
Is it documented that we will need to update this list to make sure that a connector is properly publish?

Reply:
No. This was more of a safeguard as we are migrating connectors from one flow to another. It wasn't documented since I imagined this is a short term fix until we moved everything onto the new flow and this can go away. I can document if that's useful.